ts/src/kraken.ts

```diff
@@ -1480,13 +1480,32 @@ export default class kraken extends Exchange {
             'datetime': undefined,
         };
         const currencyIds = Object.keys (balances);
+        // response currencies might be different, see comment in `fetchBalance`
+        const earningSuffix = '.F';
+        const rewardCurrenciesDict = {};
+        for (let i = 0; i < currencyIds.length; i++) {
+            const currencyId = currencyIds[i];
+            const isRewardsActivated = currencyId.endsWith (earningSuffix);
+            if (isRewardsActivated) {
+                const originalCurrencyId = currencyId.replace (earningSuffix, '');
+                rewardCurrenciesDict[originalCurrencyId] = currencyId;
+            }
+        }
         for (let i = 0; i < currencyIds.length; i++) {
             const currencyId = currencyIds[i];
-            const code = this.safeCurrencyCode (currencyId);
             const balance = this.safeValue (balances, currencyId, {});
             const account = this.account ();
             account['used'] = this.safeString (balance, 'hold_trade');
             account['total'] = this.safeString (balance, 'balance');
+            let adjustedCurrencyId = currencyId;
+            if (currencyId in rewardCurrenciesDict) {
+                adjustedCurrencyId = currencyId + '_EARNING';
```

Comment:
@ttodua do i get this correctly that a currency like "XBT.F" would be mapped to "BTC_EARNING" ? 

This seems ... utterly complicated if i'm honest.
why not leave it as "BTC.F" - so there's no mapping needed for 99% of the currencies?

"commonCurrencies" also doesn't map "XBT.M" to "BTC_FUTURES" - but to BTC.M - so i think ".F" should somewhat be treated similarily.

Reply:
no, i dont map XBT.F to BTC_EARNING. let me tell why I think this could the non-complicated solution (if we are assuming that the change only affects fetchBalance endpoint), so, per my understanding:
- as I understand, before activation, you have `SOL` key (when fetching balance)
- after activation, that `SOL` becomes `SOL.F` and the new earning balance takes up `SOL` key. (lets refer that with "new sol").
- so, I am mapping "new sol" to `SOL_EARNING` currency code, while `SOL.F` is being mapped to `SOL` currency code.
note, this happens only inside parseBalance, and I didnt touch any commoncurrencies

lmk if there is a better way.

Reply:
I'm not sure if it's possible for the same coin to be both SOL and SOL.F - but i'd assume that's possible as the raw exchange response contains both SOL and SOL.F.

so assuming the exchange would return the following: 

```
{
...
"SOL":{"balance":"20","hold_trade":"0.00000000"},
"SOL.F":{"balance":"100","hold_trade":"0.00000000"}
}
```

It's unclear what output i would expect. 
Would it be
```
{
"SOL": 20,
"SOL_EARNING": 100,
}
```

or would it be

```
{
"SOL_EARNING": 20,
"SOL": 100,
}
```

---

in my immagination, a much simpler fix would be to keep it as follows:
````
{
"SOL": 20,
"SOL.F": 100,
}
```

Now SOL is a bad example, as it's only really problematic with mapped currencies (e.g. XBT).

With that, the only "fix" would be mapping XBT.F to BTC.F (and other mapped currencies from common-currencies, obviously ...)

Reply:
lets start with `SOL` as it's simpler to show the concept. so you'd expect:
```
{
"SOL_EARNING": 20,  // SOL: 20
"SOL": 100,   // SOL.F : 100 
}
```

with XBT - BTC, let me check to think about that and i'll reply soon.

Reply:
I think keeping them as is (e.g. returning `USD.F`) is not a good idea, because users expect to get `USD` as key for their balance, independently the fact/knowing whether they had activated account rewards or not.

Reply:
things are a bit messy with XBT... they are provided with XXBT and XBT.F ... what a mess, how they could complicate things like this, none of other exchanges have such complexity.

Reply:
> what a mess, how they could complicate things like this, none of other exchanges have such complexity.

I don't feel qualified to answer this question :laughing: 

> I think keeping them as is (e.g. returning USD.F) is not a good idea, because users expect to get USD as key for their balance, independently the fact/knowing whether they had activated account rewards or not.

In principle, that's fine for me (i can remove my workaround for this).
I do think however that this decision (SOL == SOL.F) should be left to the ccxt user without inventing a new nonexistant currency.

The only thing I'm concerned about is the fact that i'm not sure if it's possible to have both SOL and SOL.F in the same account - in which case, behavior of suddenly having SOL_EARNING (which is NOT the balance i commited for earnings) is quite confusing.

Now adding insult to injury here i'll bring up another point - which you'll however probably be aware of:
if a balance is staked - that's returned as "SOL.S".

In my example, i've got FLR staked (sorry for the currency switch - i don't own SOL on kraken)

```
{
// ... 
        "FLR": {
            "balance": "0.0000",
            "hold_trade": "0.0000"
        },
        "FLR.S": {
            "balance": "72.5308",
            "hold_trade": "0.0000"
        }
}
```

Following the SOL_EARNING approach you proposed above- FLR.S would be mapped to FLR - and "FLR" to "FLR_STAKING" ? otherwise it ain't really consistant (here however i'm 100% certain that it's possible to have both balances as i can stake part of the amount).
what if there's a ".F" balance too? will i then have FLR (which one is this now, FLR.S or FLR.F), FLR_EARNING and FLR_STAKING?